common/config/kafkaConfig.go

```diff
@@ -103,7 +102,12 @@ func (k *KafkaConfig) GetTopicsForApplication(app string) TopicList {
 	return k.Applications[app]
 }
 
-// GetKafkaClusterSecureConfigForTopic gets isSecure status from topic
-func (k *KafkaConfig) GetKafkaSecureConfigForTopic(topic string) bool {
-	return k.Topics[topic].IsSecure
+// GetKafkaPropertiesForTopic gets properties from topic
+func (k *KafkaConfig) GetKafkaPropertiesForTopic(topic string) map[string]interface{} {
+	topicConfig, ok := k.Topics[topic]
+	if !ok || topicConfig.Properties == nil {
+		// No properties for the specified topic in the config
+		return nil
```

Comment:
nit: it might be better to return an empty map to avoid nil checks on the caller side. or instead of `GetKafkaPropertiesForTopic(topic string)` we can expose `GetKafkaPropertyForTopic(topic, propName string)`

Reply:
Yeah, updated to return the value for a specific property. It is probably not necessary to return the whole map.